packages/supabase/lib/src/supabase_client.dart

```diff
@@ -122,18 +133,30 @@ class SupabaseClient {
         },
         _httpClient = httpClient,
         _isolate = isolate ?? (YAJsonIsolate()..initialize()) {
-    auth = _initSupabaseAuthClient(
+    _authInstance = _initSupabaseAuthClient(
```

Comment:
When a custom access token is passed, no auth instance should be created, right? From what I see, this is still the case, but just because it's marked as `late`. I would prefer a clearer structure to ensure no auth instance is created.

Reply:
We can always come back and change the implementation of it. For now, I'm going to merge this PR as this is not really critical.